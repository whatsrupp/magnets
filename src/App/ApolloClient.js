import { ApolloClient } from 'apollo-client';
import { HttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
const GRAPHQL_API_URL = process.env.REACT_APP_GRAPHQL_API_URL
const apolloClient = new ApolloClient({
    link: new HttpLink({uri: GRAPHQL_API_URL}),
    cache: new InMemoryCache(),
})

export default apolloClient