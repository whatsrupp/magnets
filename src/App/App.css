.App{
    background-color: slateblue;
    height: 100vh;
    width: 100vw;
}