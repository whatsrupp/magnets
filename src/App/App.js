import React, { Component } from 'react';
import './App.css';
import Sidebar from 'Sidebar/Sidebar';
import PostsForm from 'PostsForm/PostsForm';
import Viewport from 'Viewport/Viewport';
import apolloClient from './ApolloClient';
import { ApolloProvider } from 'react-apollo';


class App extends Component {
  render() {
    return (
      <ApolloProvider client={apolloClient}>
        <div className="App">
          <Sidebar />
          <Viewport>
            <PostsForm />
          </Viewport>
        </div>
      </ApolloProvider>
    );
  }
}

export default App;
