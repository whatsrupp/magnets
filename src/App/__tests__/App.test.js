import React, { Component } from 'react';
import { render, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../App'


describe('<App />', ()=>{

  it('renders', ()=>{
    const wrapper = render(<App />)
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a Sidebar component', ()=>{
      const wrapper = shallow(<App />)
      const sidebar = wrapper.find('Sidebar')
      expect(sidebar.length).toEqual(1)
  })

})
