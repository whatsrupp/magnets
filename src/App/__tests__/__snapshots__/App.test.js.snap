// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "App",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Sidebar",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "logo",
              "class": "Sidebar__Logo",
              "src": "magnet.svg",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "alt": "logo",
                "class": "Sidebar__Logo",
                "src": "magnet.svg",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
          Object {
            "attribs": Object {
              "alt": "logo",
              "class": "Sidebar__Logo",
              "src": "magnet.svg",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "alt": "logo",
                "class": "Sidebar__Logo",
                "src": "magnet.svg",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
